galley/pkg/config/analysis/analyzers/testdata/namespaceconflict-reqauth-conflicts.yaml

```diff
@@ -0,0 +1,73 @@
+# Generate testing namespaces
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: foo
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: bar
+---
+apiVersion: security.istio.io/v1beta1
+kind: RequestAuthentication
+metadata:
+  name: defaultconfig
+  namespace: default
+spec:
+  jwtRules:
+  - issuer: "issuer-foo"
+---
+apiVersion: security.istio.io/v1beta1
+kind: RequestAuthentication
+metadata:
+  name: defaultconfig2
+  namespace: default    
+spec:
+  jwtRules:
+  - issuer: "issuer-bar"    # This will generate a conflict since there are two namespace wide configurations
+---
+apiVersion: security.istio.io/v1beta1
+kind: RequestAuthentication
+metadata:
+  name: fooconfig
+  namespace: foo
+spec:
+  jwtRules:
+  - issuer: "issuer-foo"  # Will not generate a conflict since it's on a new namespace 
+---
+apiVersion: security.istio.io/v1beta1
+kind: RequestAuthentication
+metadata:
+  name: seldummybar
+  namespace: bar
+spec:
+  selector:
+    matchLabels:
+      app: dummy
+  jwtRules:
+  - issuer: "issuer-foo"
+---
+apiVersion: security.istio.io/v1beta1
+kind: RequestAuthentication
+metadata:
+  name: seldummybar2
+  namespace: bar          # Will generate a conflict since there's a selector in bar for app:dummy
+spec:
+  selector:
+    matchLabels:
+      app: dummy
+  jwtRules:
+  - issuer: "issuer-bar"
+---
+apiVersion: security.istio.io/v1beta1
+kind: RequestAuthentication
+metadata:
+  name: seldummydefault
+  namespace: default
+spec:
+  selector:
+    matchLabels:
+      app: dummy          # Will generate a conflict since there's already a namespace wide configuration
```

Comment:
I don't think this should, the precedence is root namespace > namespace level > workload selector.
So its fine to have a namespace level one and a workload selector one, the latter will take precedence

Reply:
Thanks a lot for noticing that John, I think I fixed the issue, right now it's reporting the issues as follows:



For workloads that don't have a selector associated (namespace wide) (peer authentications as an example)
```
Warning [IST0146] (PeerAuthentication defaultconfig.default) More than one istio/security/v1beta1/peerauthentications in namespace "default" for workload "(ALL) Namespace: default": [default/defaultconfig default/defaultconfig2].
Warning [IST0146] (PeerAuthentication defaultconfig2.default) More than one istio/security/v1beta1/peerauthentications in namespace "default" for workload "(ALL) Namespace: default": [default/defaultconfig default/defaultconfig2].
```


And for workloads that have a selector associated (selector conflicts) (request authentications as an example)

```
Warning [IST0146] (RequestAuthentication seldummybar.bar) More than one istio/security/v1beta1/requestauthentications in namespace "bar" for workload "app=dummy": [bar/seldummybar bar/seldummybar2].
Warning [IST0146] (RequestAuthentication seldummybar2.bar) More than one istio/security/v1beta1/requestauthentications in namespace "bar" for workload "app=dummy": [bar/seldummybar bar/seldummybar2].
```

I removed that test scenario and the test cases I defined are working as expected. Please do let me know if you think I should add more validation or some other cases that I might be missing! :) 🤔